app/models/warehouse/referral_event.rb

```diff
@@ -0,0 +1,31 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/boston-cas/blob/production/LICENSE.md
+###
+
+module Warehouse
+  class ReferralEvent < Base
+    self.table_name = :cas_referral_events
+
+    belongs_to :client, foreign_key: :cas_client_id, inverse_of: :referral_events
+    belongs_to :client_opportunity_match, inverse_of: :referral_event
```

Comment:
I'm not sure exactly how this will be handled cross-db, but I think these should get `optional: true`